### PR TITLE
Split out properties from identifiers

### PR DESCRIPTION
**Pull request by lydell**
- Split out a PROPERTY token from the IDENTIFIER token.
- Split out Property from the Identifier in the grammar.
- Split out PropertyLiteral from IdentifierLiteral.

### CODE REVIEW

File: src/nodes.coffee [hunk 1]
```diff
@@ -413,6 +413,9 @@ exports.PassthroughLiteral = class PassthroughLiteral extends Literal
 exports.IdentifierLiteral = class IdentifierLiteral extends Literal
   isAssignable: -> @value not in RESERVED
 
+exports.PropertyLiteral = class PropertyLiteral extends Literal
```
Comment (michaelficarra) [inline src/nodes.coffee]: `PropertyLiteral` is a pretty strange name. I think `PropertyName` is better.

File: src/lexer.coffee [hunk 2]
```diff
@@ -114,12 +114,15 @@ exports.Lexer = class Lexer
...
+    tag =
+      if colon or prev? and
+         (prev[0] in ['.', '?.', '::', '?::'] or
+         not prev.spaced and prev[0] is '@')
+        'PROPERTY'
+      else
+        'IDENTIFIER'
 
-    if not forcedIdentifier and (id in JS_KEYWORDS or id in COFFEE_KEYWORDS)
+    if tag == 'IDENTIFIER' and (id in JS_KEYWORDS or id in COFFEE_KEYWORDS)
```
Comment (michaelficarra) [inline src/lexer.coffee]: Use `is`, not `==`.

Comment (lydell) [inline src/lexer.coffee]: Thanks. Seems like I’ve gotten too used to the style guide used in another project :) I went ahead and fixed all other `==`s and `!=`s in the diff as well.

**Comment by michaelficarra** [comment]
LGTM. Please rebase and merge.

**Pull request merged**